artifacts/deployment/02-apiserver-deployment.yaml

```diff
@@ -0,0 +1,30 @@
+# Make sure the EtcdProxy Controller and API server prerequisites are deployed before deploying the manifest.
+---
+# Creates the etcd stroage for the aggregated API server.
+# The etcd is available on http://sample-apiserver.kube-apiserver-storage.svc:2379.
+apiVersion: etcd.xmudrii.com/v1alpha1
+kind: EtcdStorage
+metadata:
+  name: kube-sample-apiserver
+---
+# Deploys the sample-apiserver (https://github.com/kubernetes/sample-apiserver).
+# The Deployment uses the xmudrii/kube-sample-apiserver image.
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: apiserver
+  metadata: kube-sample-apiserver
+spec:
+  selector:
+    matchLabels:
+      apiserver: sample-apiserver
+  template:
+    metadata:
+      labels:
+        apiserver: sample-apiserver
+    spec:
+      containers:
+      - name: sample-apiserver
+        image: xmudrii/kube-sample-apiserver
+        imagePullPolicy: Always
+        command: ["/kube-sample-apiserver", "--etcd-servers=http://wardle.kube-apiserver-storage.svc:2379"]
```

Comment:
missing the configmap and secret volume. At least put them here, even though you don't use the certs yet.

Reply:
I have created manifests for deploying the Secret, ConfigMap, and RBAC roles to ensure needed permissions for the controller SA. Also, updated the Deployment to use them.